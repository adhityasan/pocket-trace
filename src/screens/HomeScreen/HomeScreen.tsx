import { SafeAreaView, StatusBar, Text, TouchableOpacity, View } from "react-native";

import { RootStackProps } from "@@types/navigations/root";
import { SCREEN_NAMES } from "@services/constants/screen";
import globalStyles from "@styles/_global";

type HomeScreenParamsType = RootStackProps<SCREEN_NAMES.HOME>;

const HomeScreen = ({ navigation }: HomeScreenParamsType) => (
  <SafeAreaView style={[globalStyles.fullscreen, { backgroundColor: "#232323" }]}>
    <StatusBar barStyle="light-content" backgroundColor="#232323" />
    <View style={{ paddingHorizontal: 5 }}>
      <Text style={[globalStyles.titleText, { color: "#919ff0" }]}>Home Screen</Text>
      <TouchableOpacity onPress={() => navigation.navigate(SCREEN_NAMES.ABOUT)}>
        <Text style={{ color: "#eee", textAlign: "center" }}>see about screen ?</Text>
      </TouchableOpacity>
    </View>
  </SafeAreaView>
);

export default HomeScreen;
