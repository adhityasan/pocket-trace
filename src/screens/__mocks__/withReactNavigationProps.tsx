import React from "react";

export const createTestScreenProps = (props?: object) => ({
  navigation: {
    state: { params: {} },
    dispatch: jest.fn(),
    goBack: jest.fn(),
    dismiss: jest.fn(),
    navigate: jest.fn(),
    openDrawer: jest.fn(),
    closeDrawer: jest.fn(),
    toggleDrawer: jest.fn(),
    getParam: jest.fn(),
    setParams: jest.fn(),
    addListener: jest.fn(),
    push: jest.fn(),
    replace: jest.fn(),
    pop: jest.fn(),
    popToTop: jest.fn(),
    isFocused: jest.fn()
  },
  ...props
});

const withReactNavigationProps = (Component: any) => {
  const navigationProps = createTestScreenProps();

  return <Component {...navigationProps} />;
};

export default withReactNavigationProps;
