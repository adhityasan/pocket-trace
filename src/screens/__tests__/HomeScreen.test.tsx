import { create, act } from "react-test-renderer";

import HomeScreen from "@screens/HomeScreen";
import withReactNavigationProps from "@screens/__mocks__/withReactNavigationProps";

describe("screens/HomeScreen", () => {
  it("should match snapshot", () => {
    const withNavigationProps = withReactNavigationProps(HomeScreen);
    const result = create(withNavigationProps).toJSON();
    act(() => {
      expect(result).toMatchSnapshot();
    });
  });
});
