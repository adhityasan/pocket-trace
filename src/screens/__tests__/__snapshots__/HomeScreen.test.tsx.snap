// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screens/HomeScreen should match snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      },
      Object {
        "backgroundColor": "#232323",
      },
    ]
  }
>
  <View
    style={
      Object {
        "paddingHorizontal": 5,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 30,
            "fontWeight": "bold",
            "textAlign": "center",
          },
          Object {
            "color": "#919ff0",
          },
        ]
      }
    >
      Home Screen
    </Text>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#eee",
            "textAlign": "center",
          }
        }
      >
        see about screen ?
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "marginVertical": 5,
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#5145a0",
              "borderRadius": 6,
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 12,
            },
            undefined,
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#fafafa",
                "fontSize": 18,
                "fontWeight": "500",
                "textAlign": "center",
              },
              undefined,
            ]
          }
        >
          Sign out
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
